nf_core/pipelines/lint/schema_description.py

```diff
@@ -36,8 +36,9 @@ def schema_description(self):
                 warned.append(f"Ungrouped param in schema: `{up}`")
 
     # Iterate over groups and add warning for parameters without a description
-    for group_key in self.schema_obj.schema["definitions"].keys():
-        group = self.schema_obj.schema["definitions"][group_key]
+    defs_notation = self.schema_obj.defs_notation
```

Comment:
don't see that this can be any other value than `$defs` in the code

Reply:
This can still be `definitions` if the pipeline still uses nf-validation

Reply:
It's set everytime the schema of the `PipelineSchema` class is updated. This option is used heavily in `schema.py` and a bit in `schema_description.py` (or do you mean something else?)

Reply:
So it depends on which schema has been used

Reply:
I am talking about `self.schema_obj.defs_notation`. where is that value set to something different than `$defs`?

Reply:
here: https://github.com/nf-core/tools/blob/3600db75edf62709fcd48c6174c6b698c95c93f3/nf_core/pipelines/schema.py#L107

basically if `nf-validation` is set in the config, then use `definitions` else use `$defs`